.github/actions/equivalence-test/action.yml

```diff
@@ -47,15 +52,24 @@ runs:
           --binary=$(pwd)/bin/terraform
         
         changed=$(git diff --quiet -- testing/equivalence-tests/outputs || echo true)
-        if [[ $changed == "true" ]]; then
-          echo "found changes, and pushing new golden files into branch ${{ inputs.target-terraform-branch }}."
-
-          git config user.email "52939924+teamterraform@users.noreply.github.com"
-          git config user.name "The Terraform Team"
+        echo "changed=$changed" >> "${GITHUB_OUTPUT}"
 
-          git add ./testing/equivalence-tests/outputs
-          git commit -m "Automated equivalence test golden file update for release ${{ inputs.target-terraform-version }}."
-          git push
-        else
-          echo "found no changes, so not pushing any updates."
-        fi
+    - name: "branch, commit, and push changes"
+      if: steps.execute.outputs.changed == 'true'
+      shell: bash
+      run: |
+        git checkout -b ${{ inputs.new-branch }}
+        git add testing/equivalence-tests/outputs
+        git commit -m "Update equivalence test golden files."
+        git push --set-upstream origin ${{ inputs.new-branch }}
+          
+    - name: "create pull request"
```

Comment:
I think this would require

```
permissions:
  contents: write
  pull-requests: write
``` 

or is it enough to have this in the invoking github action without explicitly passing down the GITHUB_TOKEN var?

Reply:
I think because it is a composite action it inherits the permissions from the caller.